### PR TITLE
Check plugin activation error

### PR DESCRIPTION
**Pull request by Pingdred**
# Description

Check if there is an error activating the plugin and if so, return HTTP 500.

Also plugin modules are "unimported" when it is disabled.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/cat/mad_hatter/plugin.py [hunk 1]
```diff
@@ -48,15 +50,28 @@ def __init__(self, plugin_path: str, active: bool):
 
         # all plugins start active, they can be deactivated/reactivated from endpoint
         if active:
-            self.activate()
+            try:
```
Comment (pieroit) [inline core/cat/mad_hatter/plugin.py:53]: is this necessary? Having already a try/except in the `/plugins/toggle/` endpoint and inside `self._load_hooks_and_tools`?

File: core/cat/mad_hatter/plugin.py [hunk 2]
```diff
@@ -48,15 +50,28 @@ def __init__(self, plugin_path: str, active: bool):
...
+            raise e
 
     def deactivate(self):
         self._active = False
+
+        # Remove the imported modules
+        for py_file in self.py_files:
+            py_filename = py_file.replace("/", ".").replace(".py", "")  # this is UGLY I know. I'm sorry
+            log(f"Remove module {py_filename}", "DEBUG")
+            sys.modules.pop(py_filename)
```
Comment (pieroit) [inline core/cat/mad_hatter/plugin.py:73]: nice!

File: core/cat/mad_hatter/plugin.py [hunk 3]
```diff
@@ -48,15 +50,28 @@ def __init__(self, plugin_path: str, active: bool):
...
+            try:
+                self.activate()
+            except Exception as e:
+                raise e
 
     def activate(self):
-        self._active = True
-        # lists of hooks and tools
-        self._hooks, self._tools = self._load_hooks_and_tools()
+        try:
```
Comment (pieroit) [inline core/cat/mad_hatter/plugin.py:59]: is this necessary? Having already a try/except in the `/plugins/toggle/` endpoint and inside `self._load_hooks_and_tools` ?

**Comment by pieroit** [comment]
@Pingdred I'm having difficulties where the try/except blocks are necessary and where they are not.
Can we have a look together? Maybe there is something I cannot understand :)

**Comment by Pingdred** [comment]
Sure, I'll ping you on discord later 😄

**Comment by pieroit** [comment]
Thanks <3

**Pull request merged**